1 задача
 - Скачать данные по ссылке https://www.kaggle.com/datasets/esratmaria/house-price-dataset-with-other-information
 - Считать данные с помощью pandas
 - Вывести на экран первые 5 строк
 - Посмотреть на описание признаков и на их содержание

In [1]:
import pandas as pd
import numpy as np
df=pd.read_csv('//Users/efim/Desktop/GeekBrains/kc_house_data.csv')
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

In [2]:
df.tail(7)

id             date      price  bedrooms  bathrooms  \
21606  7936000429  20150326T000000  1007500.0         4       3.50   
21607  2997800021  20150219T000000   475000.0         3       2.50   
21608   263000018  20140521T000000   360000.0         3       2.50   
21609  6600060120  20150223T000000   400000.0         4       2.50   
21610  1523300141  20140623T000000   402101.0         2       0.75   
21611   291310100  20150116T000000   400000.0         3       2.50   
21612  1523300157  20141015T000000   325000.0         2       0.75   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
21606         3510      7200     2.0           0     0  ...      9   
21607         1310      1294     2.0           0     0  ...      8   
21608         1530      1131     3.0           0     0  ...      8   
21609         2310      5813     2.0           0     0  ...      8   
21610         1020      1350     2.0           0     0  ...      7   
21611         1600      2388     2.0           0     0  ...      8   
21612         1020      1076     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
21606        2600            910      2009             0    98136  47.5537   
21607        1180            130      2008             0    98116  47.5773   
21608        1530              0      2009             0    98103  47.6993   
21609        2310              0      2014             0    98146  47.5107   
21610        1020              0      2009             0    98144  47.5944   
21611        1600              0      2004             0    98027  47.5345   
21612        1020              0      2008             0    98144  47.5941   

          long  sqft_living15  sqft_lot15  
21606 -122.398           2050        6200  
21607 -122.409           1330        1265  
21608 -122.346           1530        1509  
21609 -122.362           1830        7200  
21610 -122.299           1020        2007  
21611 -122.069           1410        1287  
21612 -122.299           1020        1357  

[7 rows x 21 columns]

2 задача
 - Проведите первичный анализ данных
 - Изучите типы данных
 - Найдите количество пропущенных ячеек в данных [не найденно]
 - Посчитайте основные статистики по всем признакам и поизучайте их
 - Пишите выводы

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

In [4]:
df.describe()

id         price      bedrooms     bathrooms   sqft_living  \
count  2.161300e+04  2.161300e+04  21613.000000  21613.000000  21613.000000   
mean   4.580302e+09  5.400881e+05      3.370842      2.114757   2079.899736   
std    2.876566e+09  3.671272e+05      0.930062      0.770163    918.440897   
min    1.000102e+06  7.500000e+04      0.000000      0.000000    290.000000   
25%    2.123049e+09  3.219500e+05      3.000000      1.750000   1427.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    waterfront          view     condition  \
count  2.161300e+04  21613.000000  21613.000000  21613.000000  21613.000000   
mean   1.510697e+04      1.494309      0.007542      0.234303      3.409430   
std    4.142051e+04      0.539989      0.086517      0.766318      0.650743   
min    5.200000e+02      1.000000      0.000000      0.000000      1.000000   
25%    5.040000e+03      1.000000      0.000000      0.000000      3.000000   
50%    7.618000e+03      1.500000      0.000000      0.000000      3.000000   
75%    1.068800e+04      2.000000      0.000000      0.000000      4.000000   
max    1.651359e+06      3.500000      1.000000      4.000000      5.000000   

              grade    sqft_above  sqft_basement      yr_built  yr_renovated  \
count  21613.000000  21613.000000   21613.000000  21613.000000  21613.000000   
mean       7.656873   1788.390691     291.509045   1971.005136     84.402258   
std        1.175459    828.090978     442.575043     29.373411    401.679240   
min        1.000000    290.000000       0.000000   1900.000000      0.000000   
25%        7.000000   1190.000000       0.000000   1951.000000      0.000000   
50%        7.000000   1560.000000       0.000000   1975.000000      0.000000   
75%        8.000000   2210.000000     560.000000   1997.000000      0.000000   
max       13.000000   9410.000000    4820.000000   2015.000000   2015.000000   

            zipcode           lat          long  sqft_living15     sqft_lot15  
count  21613.000000  21613.000000  21613.000000   21613.000000   21613.000000  
mean   98077.939805     47.560053   -122.213896    1986.552492   12768.455652  
std       53.505026      0.138564      0.140828     685.391304   27304.179631  
min    98001.000000     47.155900   -122.519000     399.000000     651.000000  
25%    98033.000000     47.471000   -122.328000    1490.000000    5100.000000  
50%    98065.000000     47.571800   -122.230000    1840.000000    7620.000000  
75%    98118.000000     47.678000   -122.125000    2360.000000   10083.000000  
max    98199.000000     47.777600   -121.315000    6210.000000  871200.000000

In [5]:
df.describe(include=['object'])

date
count             21613
unique              372
top     20140623T000000
freq                142

3 задача
Ответьте на несколько вопросов
3.1 В каком диапазоне изменяются стоимости недвижимости? [- от 75 тыс. до 7.7 млн.]
3.2 Какую долю в среднем занимают жилая площадь от всей площади по всем домам? [0.14]
3.3 Как много домов с разными этажами в данных?
3.4 Насколько хорошие состояния у домов в данных?
3.5 Найдите года, когда построили первый дом, когда построили последний дом в данных? [-первый - в 1900, последний - в 2015]

In [6]:
df['price'].describe() # цена от 75 тыс. до 7.7 млн.

count    2.161300e+04
mean     5.400881e+05
std      3.671272e+05
min      7.500000e+04
25%      3.219500e+05
50%      4.500000e+05
75%      6.450000e+05
max      7.700000e+06
Name: price, dtype: float64

In [7]:
df['sqft_living'].mean()/df['sqft_lot'].mean()

0.1376781757959227

In [8]:
df['floors'].value_counts()

1.0    10680
2.0     8241
1.5     1910
3.0      613
2.5      161
3.5        8
Name: floors, dtype: int64

In [9]:
df['floors'].value_counts(normalize=True)

1.0    0.494147
2.0    0.381298
1.5    0.088373
3.0    0.028363
2.5    0.007449
3.5    0.000370
Name: floors, dtype: float64

In [10]:
df['condition'].value_counts()

3    14031
4     5679
5     1701
2      172
1       30
Name: condition, dtype: int64

In [11]:
df['yr_built'].describe() # первый построили в 1900, последний - в 2015.

count    21613.000000
mean      1971.005136
std         29.373411
min       1900.000000
25%       1951.000000
50%       1975.000000
75%       1997.000000
max       2015.000000
Name: yr_built, dtype: float64

4 задача.
Ответьте на несколько вопросов:
4.1 Сколько в среднем стоят дома, у которых 2 спальни? [401 373]
4.2 Какая в среднем общая площадь домов, у которых стоимость больше 600 000? [20 443]
4.3 Как много домов коснулся ремонт? [914]
4.4 Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4? [1 488 885]

In [12]:
df[df['bedrooms']==2]['price'].mean()

401372.681884058

In [13]:
df[df['price']>600000]['sqft_lot'].mean()

20442.524776214832

In [14]:
df[df['yr_renovated']==0]['yr_renovated'].value_counts()

0    20699
Name: yr_renovated, dtype: int64

In [15]:
df['yr_renovated'].value_counts()

0       20699
2014       91
2013       37
2003       36
2005       35
        ...  
1951        1
1959        1
1948        1
1954        1
1944        1
Name: yr_renovated, Length: 70, dtype: int64

In [24]:
df[(df['yr_renovated'] != 0)].shape[0]

914

In [17]:
df[df['grade']>10]['price'].mean() - df[df['grade']<4]['price'].mean()

1488885.1175298805

5 задача.
Ответьте на несколько вопросов:
5.1 Выберите дом клиенту:
Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента? [41]
5.2 Выберите дом клиенту:
Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома? - [415 000 : 3 000 000]
5.3 Выберите дом клиенту:
Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем? [3]

In [18]:
df1 = df[(df['waterfront']!=0) & (df['bathrooms']>=3) & (df['sqft_basement']!=0)]
df1

id             date      price  bedrooms  bathrooms  \
300    3225069065  20140624T000000  3075000.0         4       5.00   
457    2122039094  20141126T000000   705000.0         3       3.00   
1164   1247600105  20141020T000000  5110800.0         5       5.25   
1261   7636800041  20140625T000000   995000.0         3       4.50   
1315   7558700030  20150413T000000  5300000.0         6       6.00   
1361   1925069082  20150511T000000  2200000.0         5       4.25   
2626   7738500731  20140815T000000  4500000.0         5       5.50   
2864   4114601570  20141118T000000  3600000.0         3       3.25   
3914   9808700762  20140611T000000  7062500.0         5       4.50   
3934   1269200229  20140723T000000  1379900.0         3       3.25   
5760    623059016  20140717T000000  1100000.0         4       3.25   
6233   2024069008  20140619T000000  2200000.0         5       4.75   
6508   4217402115  20150421T000000  3650000.0         6       4.75   
7989   9362000040  20140623T000000  3400000.0         3       4.50   
8092   1924059029  20140617T000000  4668000.0         5       6.75   
8163   1724069079  20150319T000000  1452000.0         2       3.25   
8784   3343302110  20150306T000000  1800000.0         3       3.00   
10263   625059036  20140813T000000  2700000.0         5       4.00   
11257  3761100045  20140618T000000  3000000.0         4       4.25   
11402  3426049284  20140819T000000  2300000.0         4       3.25   
11424  5119400075  20140620T000000   950000.0         3       3.25   
11446   624069035  20141209T000000  2750000.0         4       4.00   
11844  2623039082  20150218T000000   770000.0         3       3.50   
12459  4114601580  20140724T000000  1900000.0         6       4.00   
12935  3222049044  20140612T000000   835000.0         3       3.00   
13710  2923500230  20141216T000000  2600000.0         4       4.50   
13967  3126059023  20150303T000000  3395000.0         4       3.50   
14083  3625059043  20140904T000000  3300000.0         5       4.75   
14619  7936500109  20140725T000000  2230000.0         3       3.00   
14822  3623500408  20150330T000000  2600000.0         3       3.00   
15482   624069108  20140812T000000  3200000.0         4       3.25   
15668  4030100290  20141001T000000  1680000.0         5       3.50   
16817   825059178  20140923T000000  2574000.0         4       3.75   
17767  7352200450  20150115T000000  2050000.0         4       3.25   
18199  3625059152  20141230T000000  3300000.0         3       3.25   
18455  8043700300  20140608T000000  2700000.0         4       3.25   
18556  6613000930  20140902T000000  2950000.0         4       3.25   
20325   518500480  20140811T000000  3000000.0         3       3.50   
20767  8043700105  20150417T000000  2300000.0         4       4.00   
21201   518500460  20141008T000000  2230000.0         3       3.50   
21576  9253900271  20150107T000000  3567000.0         5       4.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
300           4550     18641     1.0           1     4  ...     10   
457           1970     20978     2.0           1     3  ...      9   
1164          8010     45517     2.0           1     4  ...     12   
1261          4380     47044     2.0           1     3  ...      9   
1315          7390     24829     2.0           1     4  ...     12   
1361          4640     22703     2.0           1     4  ...      8   
2626          6640     40014     2.0           1     4  ...     12   
2864          5020     12431     2.0           1     4  ...     10   
3914         10040     37325     2.0           1     2  ...     11   
3934          3786     38038     1.0           1     4  ...      9   
5760          3190     11774     2.0           1     4  ...      8   
6233          5990     10450     2.0           1     4  ...     11   
6508          5480     19401     1.5           1     4  ...     11   
7989          5230     17826     2.0           1     4  ...     10   
8092          9640     1

In [19]:
df2 = df[((df['waterfront']!=0) | (df['view']!=0)) & (df['condition']==df['condition'].max()) & (df['yr_built']>=1980)]
df2

id             date      price  bedrooms  bathrooms  \
858     623069068  20140627T000000   425000.0         3       1.00   
3821   5452301810  20140905T000000  1575000.0         5       3.75   
6045   4045100190  20141027T000000  2196000.0         4       3.25   
8136   2597450620  20141009T000000  1511250.0         3       2.50   
8791   5067400032  20141205T000000   550000.0         3       2.50   
11257  3761100045  20140618T000000  3000000.0         4       4.25   
12273  3220079017  20150213T000000   432000.0         5       2.75   
14682  7278100515  20140821T000000  1295000.0         2       2.50   
15997  2320069260  20141027T000000   415000.0         3       2.00   
16120  2574900080  20140513T000000  1550000.0         5       3.25   
17312  3585901025  20140613T000000  1735000.0         3       2.50   
18991  7856400300  20140702T000000  1411600.0         2       2.50   
18992  7856400300  20150322T000000  1505000.0         2       2.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
858           1520    213444     1.5           0     3  ...      8   
3821          4220      9240     2.0           0     2  ...     11   
6045          4250     18000     2.0           0     3  ...     10   
8136          4010     12105     1.0           0     3  ...     11   
8791          3070     14400     1.0           0     3  ...      9   
11257         4850     12445     2.0           1     4  ...     10   
12273         2060    329903     1.5           0     3  ...      7   
14682         2910     19449     2.0           1     4  ...      9   
15997         2010     33090     1.5           0     2  ...      8   
16120         3370     17458     1.0           0     2  ...     10   
17312         4310     32093     1.5           0     4  ...     10   
18991         3180      9400     2.0           0     4  ...     10   
18992         3180      9400     2.0           0     4  ...     10   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
858          1520              0      1988             0    98027  47.5081   
3821         3420            800      1991             0    98040  47.5895   
6045         3350            900      1980             0    98040  47.5612   
8136         2600           1410      1983             0    98006  47.5540   
8791         1720           1350      1985             0    98198  47.3716   
11257        3850           1000      1989             0    98034  47.7011   
12273        2060              0      1989             0    98022  47.1776   
14682        1940            970      1985             0    98177  47.7729   
15997        2010              0      1986             0    98022  47.2133   
16120        2000           1370      1982             0    98040  47.5591   
17312        2510           1800      1982             0    98177  47.7624   
18991        2610            570      1985             0    98006  47.5617   
18992        2610            570      1985             0    98006  47.5617   

          long  sqft_living15  sqft_lot15  
858   -122.093           2640      213444  
3821  -122.229           3380        9240  
6045  -122.229           3790       14537  
8136  -122.151           4010       15081  
8791  -122.321           2020       18211  
11257 -122.244           3350       12210  
12273 -121.944           2240      220232  
14682 -122.393           2540       23598  
15997 -122.007           1840       22620  
16120 -122.229           4240       15202  
17312 -122.380           3810       26400  
18991 -122.158           3760        9450  
18992 -122.158           3760        9450  

[13 rows x 21 columns]

In [20]:
df2['price'].describe()

count    1.300000e+01
mean     1.353912e+06
std      7.608267e+05
min      4.150000e+05
25%      5.500000e+05
50%      1.505000e+06
75%      1.575000e+06
max      3.000000e+06
Name: price, dtype: float64

In [21]:
df3 = df[(df['sqft_basement']==0) & (df['floors']==2) & (df['price']<=150000)]
df3

id             date     price  bedrooms  bathrooms  \
2589   5061300030  20140508T000000  134000.0         2       1.50   
10105  5466310060  20150324T000000  139500.0         2       1.50   
11255  1843200350  20140722T000000  150000.0         2       1.50   
15610  7614100080  20150211T000000  140000.0         3       1.75   
18332  9272202260  20140924T000000  130000.0         3       1.00   
18939  4239400300  20141129T000000   90000.0         3       1.00   
18988  7399100210  20141126T000000  140000.0         3       1.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
2589           980      5000     2.0           0     0  ...      7   
10105         1230      1561     2.0           0     0  ...      7   
11255         1360      1934     2.0           0     0  ...      7   
15610         1270      8991     2.0           0     0  ...      7   
18332         1200      7000     2.0           0     0  ...      7   
18939          980      2490     2.0           0     0  ...      6   
18988         1200      2002     2.0           0     0  ...      8   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
2589          980              0      1922          2003    98014  47.7076   
10105        1230              0      1983             0    98042  47.3768   
11255        1360              0      1978             0    98092  47.2857   
15610        1270              0      1981             0    98042  47.3563   
18332        1200              0      1908             0    98116  47.5883   
18939         980              0      1969             0    98092  47.3170   
18988        1200              0      1966             0    98055  47.4659   

          long  sqft_living15  sqft_lot15  
2589  -121.359           1040        5000  
10105 -122.149           1660        2243  
11255 -122.189           1360        1898  
15610 -122.149           1270        8993  
18332 -122.384           3290        6000  
18939 -122.182            980        3154  
18988 -122.189           1270        1848  

[7 rows x 21 columns]

In [22]:
df3['condition'].describe()

count    7.0
mean     3.0
std      1.0
min      1.0
25%      3.0
50%      3.0
75%      3.5
max      4.0
Name: condition, dtype: float64

In [23]:
df3['condition'].mean()

3.0